### PR TITLE
provider/aws: New DataSource: aws_elb_hosted_zone_id

### PR DESCRIPTION
**Pull request by stack72**
This datasource is a list of all of the ELB DualStack Hosted Zone IDs.
This will allow us to reference the correct hosted zone id when creating
route53 alias records

There are many bugs for this - this is just the beginning of fixing them

```
% make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSElbHostedZoneId_basic'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/01/04 13:04:32 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSElbHostedZoneId_basic -timeout 120m
=== RUN   TestAccAWSElbHostedZoneId_basic
--- PASS: TestAccAWSElbHostedZoneId_basic (20.46s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/aws	20.484s
```

### CODE REVIEW

**Comment by stack72** [comment]
//cc @vancluever :)

**Comment by vancluever** [comment]
Nice!

File: builtin/providers/aws/data_source_aws_elb_hosted_zone_id.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	"ca-central-1":   "ZQSVJUPU6J1EY",
+	"eu-central-1":   "Z215JYRZR1TBD5",
+	"eu-west-1":      "Z32O12XQLNTSW2",
+	"eu-west-2":      "ZHURV8PSTC4K8",
+	"us-east-1":      "Z35SXDOTRQ7X7K",
+	"us-east-2":      "Z3AADJGX6KTTL2",
+	"us-west-1":      "Z368ELLRRE2KJ0",
+	"us-west-2":      "Z1H1FL5HABSF5",
+	"sa-east-1":      "Z2P70J7HTTTPLU",
+}
```
Comment (ryansch) [inline builtin/providers/aws/data_source_aws_elb_hosted_zone_id.go]: For future reference this list is published by AWS at https://docs.aws.amazon.com/general/latest/gr/rande.html#elb_region

**Comment by martijnvermaat** [comment]
@stack72 Shouldn't the docs for `aws_elb` attribute `zone_id` be updated to reflect this? They say:
> to be used in a Route 53 Alias record

Or perhaps that attribute should be removed/deprecated altogether?

**Pull request merged**